raft/src/test/java/org/apache/kafka/raft/KafkaRaftClientSnapshotTest.java

```diff
@@ -276,16 +281,16 @@ public void testListenerRenotified(boolean withKip853Rpc) throws Exception {
     }
 
     @ParameterizedTest
-    @ValueSource(booleans = { false, true })
-    public void testLeaderImmediatelySendsSnapshotId(boolean withKip853Rpc) throws Exception {
+    @EnumSource(value = RaftProtocol.class)
```

Comment:
we should have probably picked the same strategy for KafkaRaftClientSnapshotTest and KafkaRaftClientTest (either test all protocols for both tests or directly translate to matrix of KIP_595 and KIP_853)

I'm not sure if it's a big enough deal to ask you to standardize though - @jsancio thoughts?

Reply:
I agree that we should try to be consistent. I prefer if we inject the `RaftProtocol` directly to the test as it is the most flexible abstraction. And to not inject a boolean that then gets converted to a `RaftProtocol`.

I am okay testing all possible values of `RaftProtocol` as long as it is not too much work to support and the tests don't take a lot longer.